xmodule/library_content_block.py

```diff
@@ -264,6 +242,108 @@ def _publish_event(self, event_name, result, **kwargs):
         self.runtime.publish(self, f"edx.librarycontentblock.content.{event_name}", event_data)
         self._last_event_result_count = len(result)  # pylint: disable=attribute-defined-outside-init
 
+    def available_children(self) -> list[tuple[str, str]]:
+        """
+        Returns (block_type, block_id) tuples which children are possible for selection,
+        in the original order from the candidates list (if manual) or lib-copied children list (if not).
+        """
+        return self._available_children(
+            library_children=self.children,
+            candidates=list(map(tuple, self.candidates)),  # Convert from nested list to list-of-tuples.
+            manual=self.manual,
+        )
+
+    @classmethod
+    def _available_children(
```

Comment:
This was previously "_get_valid_children". I renamed it "_available_children" because it struck me as more self-explanatory -- these are the children which are available for selection.